### PR TITLE
Implement assert_scalar_eq!

### PR DESCRIPTION
**Pull request by Andlon**
This PR provides an implementation of a `assert_scalar_eq!` macro which works analogously to `assert_matrix_eq!` and `assert_vector_eq!`.

Currently, we have a bunch of tests for scalars that look roughly like this:

```
let det = lu.det();
let expected_det = 3.0;
let diff = det - expected_det;
assert!(diff.abs() < 1e-6);
```

This is less than ideal for several reasons. For one, if the test fails, the test does not report what caused it to fail. Moreover, the lack of a standardized procedure means that the requirements are often arbitrary. The macro provided in this PR lets us instead do

```
assert_scalar_eq!(lu.det(), 3.0, comp = float);
```

Here we use the float comparator, which provides reasonable defaults, and the tolerance used (ULP and absolute epsilon value) can be tweaked if necessary. Of course, the other comparators (`exact`, `abs` and `ulp`) are also available.

EDIT: This addresses one of the points mentioned in #95.

### CODE REVIEW

**Comment by AtheMathmo** [comment]
Looks good to me!

From what I can see this PR does not address areas of the PR where we want to use this new macro. I think we should spin this out as a new issue (tagged easy) and hopefully somebody else picks it up! :)

File: src/macros/assert_scalar_eq.rs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+/// assert_scalar_eq!(x, y, comp = abs, tol = 0.1);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! assert_scalar_eq {
+    ($x:expr, $y:expr) => {
+        {
+            // Note: The reason we take slices of both x and y is that if x or y are passed as references,
+            // we don't attempt to call elementwise_matrix_comparison with a &&BaseMatrix type (double reference),
+            // which does not work due to generics.
```
Comment (Andlon) [inline src/macros/assert_scalar_eq.rs]: Some copypasta comment here which doesn't make sense in the scalar situation

**Comment by Andlon** [comment]
@AtheMathmo: I agree with the point about documentation. I will change this.

**Comment by Andlon** [comment]
OK, I added some very brief notes on where to find details about floating-point comparison issues and how to resolve them with the `assert_matrix/vector/scalar_eq` macros. I guess that should do it?

**Comment by AtheMathmo** [comment]
Look good to me. Merging now and will open a new issue for using these macros.

**Pull request merged**